source/common/upstream/health_checker_base_impl.cc

```diff
@@ -419,6 +420,26 @@ HealthCheckerImplBase::ActiveHealthCheckSession::clearPendingFlag(HealthTransiti
 }
 
 void HealthCheckerImplBase::ActiveHealthCheckSession::onIntervalBase() {
+  /*Only if host is healthy, check whether need idle hc*/
+  if (parent_.enable_idle_hc_ && !host_->healthFlagGet(Host::HealthFlag::FAILED_ACTIVE_HC)) {
+    MonotonicTime traffic_pass_time;
+    switch(parent_.healthCheckerType()) {
+      case envoy::data::core::v3::HealthCheckerType::TCP:
+        traffic_pass_time = host_->lastTrafficPassTime();
+        break;
+      case envoy::data::core::v3::HealthCheckerType::HTTP:
+        traffic_pass_time = host_->lastTrafficPassTime2xx();
+        break;
+      default:
+        break;
```

Comment:
What about gRPC, etc.? We should handle the cases for all health checkers if possible. Redis, etc.